src/main/java/edu/wpi/grip/core/operations/composite/HSLThresholdOperation.java

```diff
@@ -0,0 +1,91 @@
+            () -> Arrays.asList(0.0, 180.00), SocketHint.View.RANGE, new List[]{Arrays.asList(0.0, 180.0)});
+    private final SocketHint<List> saturationHint = new SocketHint<List>("Saturation", List.class,
+            () -> Arrays.asList(0.0, 255.0), SocketHint.View.RANGE, new List[]{Arrays.asList(0.0, 255.0)});
+    private final SocketHint<List> luminanceHint = new SocketHint<List>("Luminance", List.class,
+            () -> Arrays.asList(0.0, 255.0), SocketHint.View.RANGE, new List[]{Arrays.asList(0.0, 255.0)});
+
+    private final SocketHint<opencv_core.Mat> outputHint = new SocketHint<opencv_core.Mat>("output", opencv_core.Mat.class, opencv_core.Mat::new);
+
+    @Override
+    public String getName() {
+        return "HSL Threshold";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Segment an image based on hue, saturation and luminance ranges";
+    }
+
+    @Override
+    public InputSocket<?>[] createInputSockets(EventBus eventBus) {
+        return new InputSocket<?>[]{
+                new InputSocket<>(eventBus, inputHint),
+                new InputSocket<>(eventBus, hueHint),
+                new InputSocket<>(eventBus, saturationHint),
+                new InputSocket<>(eventBus, luminanceHint),
+        };
+    }
+
+    @Override
+    public OutputSocket<?>[] createOutputSockets(EventBus eventBus) {
+        return new OutputSocket<?>[]{
+                new OutputSocket<>(eventBus, outputHint)
+        };
+    }
+
+    @Override
+    public void perform(InputSocket<?>[] inputs, OutputSocket<?>[] outputs) {
+        final opencv_core.Mat input = ((InputSocket<opencv_core.Mat>) inputs[0]).getValue();
+        final List<Number> channel1 = ((InputSocket<List<Number>>) inputs[1]).getValue();
+        final List<Number> channel2 = ((InputSocket<List<Number>>) inputs[2]).getValue();
+        final List<Number> channel3 = ((InputSocket<List<Number>>) inputs[3]).getValue();
+
+        final OutputSocket<opencv_core.Mat> outputSocket = (OutputSocket<opencv_core.Mat>) outputs[0];
+        final opencv_core.Mat output = outputSocket.getValue();
+
+        // Do nothing if nothing is connected to the input
+        if (input.empty()) {
+            // TODO: Indicate input invalid
+            return;
+        }
+
+        // Intentionally 1, 3, 2. This maps to the HLS open cv expects
+        final opencv_core.Mat low = new opencv_core.Mat(input.size(), input.type(), new opencv_core.Scalar(
+                channel1.get(0).doubleValue(),
+                channel3.get(0).doubleValue(),
+                channel2.get(0).doubleValue(), 0));
+
+        final opencv_core.Mat high = new opencv_core.Mat(input.size(), input.type(), new opencv_core.Scalar(
+                channel1.get(1).doubleValue(),
+                channel3.get(1).doubleValue(),
+                channel2.get(1).doubleValue(), 0));
+
+        final opencv_core.Mat hls = new opencv_core.Mat();
```

Comment:
This is probably just something intellij automatically messed up when you copied and pasted, but open_core.* is usually statically imported.